src/DSManager.cpp

```diff
@@ -481,6 +643,68 @@ types::DynamicPubSubType* DSManager::setType(
     return pDt;
 }
 
+template<>
+void DSManager::getPubSubEntityFromParticipantGuid< eprosima::fastdds::dds::DataWriter >(
+        GUID_t& id,
+        DomainEntity*& pubsub)
+{
+    std::lock_guard<std::recursive_mutex> lock(management_mutex);
+    pubsub = entity_map[id].publisher;
+}
+
+template<>
+void DSManager::getPubSubEntityFromParticipantGuid< eprosima::fastdds::dds::DataReader >(
+        GUID_t& id,
+        DomainEntity*& pubsub)
+{
+    std::lock_guard<std::recursive_mutex> lock(management_mutex);
+    pubsub = entity_map[id].subscriber;
+}
+
+void DSManager::setParticipantInfo(
+        GUID_t guid,
+        ParticipantCreatedEntityInfo info)
+{
+    std::lock_guard<std::recursive_mutex> lock(management_mutex);
+    entity_map[guid] = info;
+}
+
+void DSManager::setParentGUID(
+        GUID_t parent,
+        GUID_t child)
+{
+    std::lock_guard<std::recursive_mutex> lock(management_mutex);
+    guid_map[child] = parent;
+}
+
+GUID_t DSManager::getParentGUID(
+        GUID_t child)
+{
+    std::lock_guard<std::recursive_mutex> lock(management_mutex);
+    return guid_map[child];
+}
+
+void DSManager::setParticipantTopic(
+        DomainParticipant* p,
+        Topic* t)
+{
+    std::lock_guard<std::recursive_mutex> lock(management_mutex);
+    entity_map[p->guid()].registeredTopics[t->get_name()] = t;
+}
+
+Topic* DSManager::getParticipantTopicByName(
+        DomainParticipant* p,
+        std::string name)
+{
+    Topic* returnTopic = nullptr;
+    std::lock_guard<std::recursive_mutex> lock(management_mutex);
+
+    returnTopic = entity_map[p->guid()].registeredTopics[name];
+
+
+    return returnTopic;
+}
```

Comment:
Being public API we should check the validity of the given pointers.

Also, it is assumed that the element `entity_map[p->guid()]` exists. Is it possible that it does not? If so, we should also check it.